### DISCUSSION TITLE
[Kavita] JPEG XL support

### DISCUSSION THREAD

**Discussion by sammy-ette** [Kareadita/Kavita#2610] [Ideas]
### Idea Description

https://en.wikipedia.org/wiki/JPEG_XL

Advantages: it's smaller
> Lossy: about 60% smaller than JPEG for same visual quality
> Lossless JPEG transcoding: about 20% smaller
> Lossless: about 35% smaller than PNG (50% smaller for HDR)

"Backwards compatible with JPEG"

Disadvantages:
Support: Chrome and Firefox require an addon for support
????

### Duration of Using Kavita

many moons

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by Nebelherr23** [comment]
I currently convert my western Comics to WebP. Due to size limitations, I cannot do the same to webcomics, where the total pixel count may not exceed 16383x16383, which makes the conversion a challenge... JXL does not have these limitations, I will post results of a few experiments regarding size at various qualities.

**Reply by scare376** [reply]
This is true but the support for JXL isn't there. This isn't being considered until more browsers support it out of the box. https://caniuse.com/jpegxl

**Reply by Cducharme84** [reply]
> I currently covert my western Comics to WebP. Due to size limitations, I cannot do the same to webcomics, where the total pixel count may not exceed 16383x16383, which makes the conversion a challenge... JXL does not have these limitations, I will post results of a few experiments regarding size at various qualities.

Have you tried avif? It’s of similar compression to webp, has support with advanced profile for 16384x8704 for single images which can be further expanded by grid derivation (essentially tiling). I myself just use webp as I read the western books, but could see avif being useful for webtoons.

**Reply by Nebelherr23** [reply]
> This is true but the support for JXL isn't there. This isn't being considered until more browsers support it out of the box. https://caniuse.com/jpegxl

I am aware, I am still doing the research, so it's available, when this is being considered.


> Have you tried avif?

No, not yet. But this is a discussion for Discord, I guess.

**Comment by BPplays** [comment]
@majora2007 would you accept a PR by someone that adds jxl support for just the reader and generating covers from (not into jxl from a jxl).
i think supporting reading jxl would be nice since mihon supports jxl, especially since i would guess most users are probably the only user in their instance so they could only mess it up for themselves

**Reply by majora2007** [reply]
I would not. As mentioned, browsers don't support. It will lead to a bad UX. Having any jxl logic in the app balloons implementation and I am now the owner of any maintenance or bugs. 

When chrome and FF add support, I will as well.

**Reply by sammy-ette** [reply]
jpeg xl support in browsers is dead in the water. chrome isn't going to support it and i have no idea about firefox.

**Reply by sammy-ette** [reply]
there is still the extension though, so it's not like using it is impossible.

me personally, @BPplays if you or someone else makes a fork i would use it.

**Reply by jwillikers** [reply]
JPEG-XL adoption is slowly gaining traction. Don't know how long we'll have to wait for the politics to play out in Chrome / Firefox, but I bet they'll eventually support it. Here's a somewhat recent article on the subject from Phoronix: https://www.phoronix.com/news/Mozilla-Interest-JPEG-XL-Rust. Kinda funny, right?

**Reply by zhjihuang** [reply]
Some news, Chromium had recently [merged](https://chromium-review.googlesource.com/c/chromium/src/+/7184969) jxl support and Chrome beta version 145.0.7632.26 supports jxl via the #enable-jxl-image-format flag. I'm sure Firefox won't be far behind as they are looking to use the same rust based decoder.

**Reply by scare376** [reply]
firefox has had it available as a flag in its nightly for a while now it seems. https://caniuse.com/jpegxl 
edit - that seems to still be the old decoder at this time. they're still working on implementing the rust decoder

**Reply by zhjihuang** [reply]
Another update: Firefox 152 just released with with jxl support by default. I think that makes jxl compatible in both Firefox and Chrome now.

**Reply by KucharczykL** [reply]
The support for JPEG-XL is _built-in_ by default but still not enabled by default. Confusing wording.

<img width="2340" height="736" alt="image" src="https://github.com/user-attachments/assets/1e963739-9bbc-469e-adcb-23aeb9b96013" />

**Comment by kingvall** [comment]
I’m really looking forward to JPEG XL support, as more and more browsers are starting to adopt it; at this point, the lack of JXL support is the last thing preventing me from fully switching to Kavita.
